src/cmd/builtin/relay.c

```diff
@@ -131,8 +131,18 @@ static int cmd_relay (optparse_t *p, int ac, char *av[])
     flux_t *h;
     int optindex;
     char *uri;
+    char hostname [MAXHOSTNAMELEN + 1];
 
-    log_init ("flux-relay");
+    /*  If possible, initalize logging prefix as local hostname.
+     *
+     *  This will be more helpful than a literal "flux-relay" logging prefix
+     *  for end users that may be uknowningly using flux-relay as part of
+     *  the ssh connector.
+     */
+    if (gethostname (hostname, MAXHOSTNAMELEN) < 0)
```

Comment:
By my read of [gethostname(1)](https://man7.org/linux/man-pages/man2/gethostname.2.html) the length parameter should be the size of the buffer not the maximum string length, so use MAXHOSTNAMELEN + 1, or maybe better, `sizeof (hostname)` here?

Also, not sure if this is important but the man page references HOST_NAME_MAX not MAXHOSTNAME.  (Might be an alias but the former is apparently the POSIX one).  Looking around our code base we use both :-(

Another thought is to use "unknown-host" or similar as the fallback if `gethostname()` fails (very unlikely).  Could just statically initialize hostname[] to that value and avoid the duplicate call to `log_init()`.

Reply:
Yeah, I did also see the `HOST_NAME_MAX` reference in the man page as well, but when I spot checked our codebase I saw use of `MAXHOSTNAMELEN`. Rather than fix it now, I chose `MAXHOSTNAMELEN` for consistency (though I guess I missed the uses of `HOST_NAME_MAX`)